### PR TITLE
ISSUE-1129 Configure schema name and version to compute schema for HDFS source.

### LINKED ISSUE #1129: Configure schema name and version to compute schema for HDFS source.

### PR DESCRIPTION
**Pull request by satishd**
Closes #1129

### CODE REVIEW

File: bootstrap/components/sources/hdfs-source-topology-component.json [hunk 1]
```diff
@@ -33,6 +33,24 @@
...
+        "hint": "schema"
+      },
+      {
+        "uiName": "Schema Version",
+        "fieldName": "schemaVersion",
+        "isOptional": false,
+        "tooltip": "Version of the given schema",
+        "type": "number",
+        "defaultValue": "-1",
+        "hint": "schema"
```
Comment (priyank5485) [inline bootstrap/components/sources/hdfs-source-topology-component.json]: Can we change schemaVersion field to match what we have in kafka source? Mainly type and hint

Comment (satishd) [reply bootstrap/components/sources/hdfs-source-topology-component.json]: Made the changes as suggested.

File: bootstrap/components/sources/hdfs-source-topology-component.json [hunk 2]
```diff
@@ -33,6 +33,24 @@
...
         "hint": "schema"
       },
+      {
+        "uiName": "Schema Name",
+        "fieldName": "schemaName",
+        "isOptional": false,
+        "tooltip": "Name of the schema to which messages in the given `Source Dir` adhere to.",
+        "type": "string",
+        "defaultValue": "",
+        "hint": "schema"
```
Comment (shahsank3t) [inline bootstrap/components/sources/hdfs-source-topology-component.json]: @satishd 
You also need to change `type`, `defaultValue` and `hint` here as done in Kafka-source.
https://github.com/hortonworks/streamline/blob/master/bootstrap/components/sources/kafka-source-topology-component.json#L33-L35

File: bootstrap/components/sources/hdfs-source-topology-component.json [hunk 3]
```diff
@@ -33,6 +33,24 @@
         "defaultValue": "/tmp/sourcefiles/",
         "hint": "schema"
```
Comment (shahsank3t) [inline bootstrap/components/sources/hdfs-source-topology-component.json]: @satishd I assume the user will be using newly added `Schema Name` field to get schema data so I assume this hint is no longer needed.

@priyank5485 Your thoughts?

File: bootstrap/components/sources/hdfs-source-topology-component.json (hunk 2 above)
Comment (satishd) [reply bootstrap/components/sources/hdfs-source-topology-component.json]: @shahsank3t Why does it need to be enumstring? Here, schema name should be given by user as a string and it is not populated from other fields like kafka source.

File: bootstrap/components/sources/hdfs-source-topology-component.json (hunk 3 above)
Comment (satishd) [reply bootstrap/components/sources/hdfs-source-topology-component.json]: @shahsank3t I am trying to understand the existing config here. Does that indicate to get schema using that name from registry?

File: bootstrap/components/sources/hdfs-source-topology-component.json (hunk 2 above)
Comment (shahsank3t) [reply bootstrap/components/sources/hdfs-source-topology-component.json]: @satishd Sorry for the confusion. Yes, we don't need to change this. My bad.

File: bootstrap/components/sources/hdfs-source-topology-component.json (hunk 3 above)
Comment (shahsank3t) [reply bootstrap/components/sources/hdfs-source-topology-component.json]: @satishd As discussed/explained earlier, you need to remove hint from here, rest of the changes looks good.

**Comment by shahsank3t** [comment]
@satishd 
+1 once you remove the hint for "SourceDir" field.

Tested out this PR and it works as follows:
1. Source Dir can be anything entered by the user.
2. User needs to enter Schema Name, UI makes a call to SR and populates version in Schema Version field.
3. User selects one of the version and UI makes another call to SR to get schema and show in SAM.

**Comment by satishd** [comment]
@shahsank3t Thanks for trying this out. I have made that change in latest commit.

**Pull request closed without merging**